pkg/backporter/backporter.go

```diff
@@ -553,30 +558,118 @@ func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String, m
 			return
 		}
-		targetRelease := bugzilla.BugUpdate{
-			TargetRelease: []string{
-				req.FormValue("release"),
-			},
+		sort.SliceStable(clones, func(i, j int) bool {
+			if len(clones[i].TargetRelease) == 0 && len(clones[j].TargetRelease) == 0 {
+				return false
+			} else if len(clones[i].TargetRelease) == 0 {
+				return true
+			} else if len(clones[j].TargetRelease) == 0 {
+				return false
+			}
+			return clones[i].TargetRelease[0] < clones[j].TargetRelease[0]
+		})
+		toCloneMajorMinorRelease, err := getMajorMinorRelease(req.FormValue("release"))
+		if err != nil {
+			handleError(w, err, fmt.Sprintf("%s cannot be parsed as release, must be of the form X.X.X", req.FormValue("release")), http.StatusBadRequest, endpoint, bugID, m)
+			return
+		}
+		var sourceBug *bugzilla.Bug
+		var sourceBugMajorMinorRel string
+		for _, clone := range clones {
+			if len(clone.TargetRelease) == 0 || clone.TargetRelease[0] == "---" {
+				continue
+			}
+			cloneMajorMinorRel, err := getMajorMinorRelease(clone.TargetRelease[0])
+			if err != nil {
+				handleError(w, err, fmt.Sprintf("%s cannot be parsed as release, must be of the form X.X.X", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if toCloneMajorMinorRelease == cloneMajorMinorRel {
+				handleError(w, err, fmt.Sprintf("clone for major release %s already exists", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if cloneMajorMinorRel > toCloneMajorMinorRelease {
+				sourceBug = clone
+				sourceBugMajorMinorRel = cloneMajorMinorRel
+				break
+			}
 		}
-		// Updating the cloned bug with the right target version
-		if err := client.UpdateBug(cloneID, targetRelease); err != nil {
-			handleError(w, err, fmt.Sprintf("clone created - Bug#%d, but failed to specify version for the cloned bug.", cloneID), http.StatusInternalServerError, endpoint, bugID, m)
+		if sourceBug == nil {
+			handleError(w, err, fmt.Sprintf("one bug with greater release needs to be present to clone from (highest release present is %s)", clones[len(clones)-1].TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
 			return
 		}
+		var sortedMajorRelease []string
+		for release := range allTargetVersions {
+			mRel, err := getMajorMinorRelease(release)
+			if err != nil {
+				handleError(w, err, fmt.Sprintf("%s cannot be parsed as release, must be of the form X.X.X", req.FormValue("release")), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			sortedMajorRelease = append(sortedMajorRelease, mRel)
+		}
+		// Sort the releases in descending order
+		sort.SliceStable(sortedMajorRelease, func(i, j int) bool {
+			return sortedMajorRelease[i] > sortedMajorRelease[j]
+		})
+		var targetRelease int
+		for ind, rel := range sortedMajorRelease {
+			if rel == sourceBugMajorMinorRel {
+				targetRelease = ind + 1
+				break
+			}
+		}
+		newClones := []string{}
```

Comment:
just use `var newClones []string`, no need to initialize the var, `append` will do that when needed